components/mediation/inbound-endpoints/org.wso2.micro.integrator.inbound.endpoint/src/main/java/org/wso2/carbon/inbound/endpoint/protocol/cdc/CDCPollingConsumer.java

```diff
@@ -0,0 +1,121 @@
+import java.util.Properties;
+import java.util.concurrent.ConcurrentLinkedQueue;
+import java.util.concurrent.ExecutorService;
+
+import static org.wso2.carbon.inbound.endpoint.protocol.cdc.CDCProcessor.inboundEpEventQueueMap;
+
+/**
+ * This class implement the processing logic related to inbound CDC protocol.
+ * Common functionalities are include in synapse
+ * util that is found in synapse commons
+ */
+public class CDCPollingConsumer {
+
+    private static final Log logger = LogFactory.getLog(CDCPollingConsumer.class);
+    private Properties cdcProperties;
+    private ExecutorService executorService = null;
+    private String inboundEndpointName;
+    private SynapseEnvironment synapseEnvironment;
+    private long scanInterval;
+    private Long lastRanTime;
+    private CDCInjectHandler injectHandler;
+
+    public CDCPollingConsumer(Properties cdcProperties, String inboundEndpointName, SynapseEnvironment synapseEnvironment,
+                              long scanInterval) {
+        this.cdcProperties = cdcProperties;
+        this.inboundEndpointName = inboundEndpointName;
+        this.synapseEnvironment = synapseEnvironment;
+        this.scanInterval = scanInterval;
+        this.lastRanTime = null;
+    }
+
+    /**
+     * Register a handler to process the file stream after reading from the
+     * source
+     *
+     * @param injectHandler
+     */
+    public void registerHandler(CDCInjectHandler injectHandler) {
+        this.injectHandler = injectHandler;
+    }
+
+    /**
+     * This will be called by the task scheduler. If a cycle execution takes
+     * more than the schedule interval, tasks will call this method ignoring the
+     * interval. Timestamp based check is done to avoid that.
+     */
+    public void execute() {
+        try {
+            if (logger.isDebugEnabled()) {
+                logger.debug("Start : CDC Inbound EP : " + inboundEndpointName);
+            }
+            // Check if the cycles are running in correct interval and start
+            // scan
+            long currentTime = (new Date()).getTime();
+            if (lastRanTime == null || ((lastRanTime + (scanInterval)) <= currentTime)) {
+                lastRanTime = currentTime;
+                poll();
+            } else if (logger.isDebugEnabled()) {
+                logger.debug(
+                        "Skip cycle since cuncurrent rate is higher than the scan interval : CDC Inbound EP : " + inboundEndpointName);
+            }
+            if (logger.isDebugEnabled()) {
+                logger.debug("End : CDC Inbound EP : " + inboundEndpointName);
+            }
+        } catch (Exception e) {
+            logger.error("Error while getting events. " + e.getMessage(), e);
+        }
+    }
+
+    /**
+     * Do the CDC processing operation for the given set of properties. Then inject
+     * according to the registered handler
+     */
+    public ChangeEvent<String, String> poll() {
+
+        if (logger.isDebugEnabled()) {
```

Comment:
Since we don't have string concatenations in the log message, we can skip this `logger.isDebugEnabled()` check as `logger.debug` has it's own check.

Reply:
yes this was fixed